### DISCUSSION TITLE
Should @defer & @stream be supported on mutations and subscriptions?

### DISCUSSION THREAD

**Discussion by robrichard** [graphql/defer-stream-wg#2] [Architectural Decision Records]
## Context
It is often asked if it is necessary to support `@defer` & `@stream` on mutations. 

## Decision
We believe this is an essential feature.

## Consequences
In Relay it is common to write mutations and subscriptions that reuse fragments that are coupled to UI components. It would be difficult to ensure that any fragment that is spread in a mutation/subscription does not include any other fragment that is deferred or any field that is streamed. It is also desirable for these fields to be returned asynchronously for the same reasons they are deferred or streamed on a query operation.

## Status
- [x] defined in spec
- [x] implemented in graphql-js

## References
* #1

**Comment by glasser** [comment]
Is there a separate discussion of defer/stream at the *top level* of mutations and subscriptions?

**Reply by glasser** [reply]
Ah found it: #19
